app/src/main/java/com/pajato/android/gamechat/common/InvitationManager.java

```diff
@@ -17,35 +17,185 @@
+    public void onAuthenticationChange(final AuthenticationChangeEvent event) {
+        Account account = event != null ? event.account : null;
+        if (account == null) return;
+
+        // Update Firebase
+        boolean accountChanged = false;
+        for (String key : mInvitedGroups.keySet()) {
+            // If the account has already joined, don't add it again!
+            if (account.joinList.contains(key)) {
+                mInvitedGroups.remove(key);
+            } else {
+                account.joinList.add(key);
+                accountChanged = true;
+                mInvitedGroups.get(key).addedToAccountJoinList = true;
+                // Set up the watcher for the common room (the group watcher is set up by
+                // adding the group to the account joinList).
+                RoomManager.instance.setWatcher(key, mInvitedGroups.get(key).commonRoomKey);
+            }
+        }
+        if (accountChanged) {
+            AccountManager.instance.updateAccount(account);
+        }
+    }
+
+    /** Handle the room profile change */
+    @Subscribe public void onRoomProfileChange(@NonNull final ProfileRoomChangeEvent event) {
+        // If this room is one we care about, add the current account to the room memberIdList.
+        Account currAccount = AccountManager.instance.getCurrentAccount();
+        for (Map.Entry<String, GroupInviteData> entry : mInvitedGroups.entrySet()) {
+            GroupInviteData data = entry.getValue();
+            if (data.commonRoomKey.equals(event.key)) {
+                event.room.memberIdList.add(currAccount.id);
+                RoomManager.instance.updateRoomProfile(event.room);
+                data.addedToCommRoomMemberList = true;
+                mInvitedGroups.put(entry.getKey(), data);
+
+                // Post a message to the common room announcing the user has joined
+                String format = messageMap.get(Integer.valueOf(R.string.HasJoinedMessage));
+                String text = String.format(Locale.getDefault(), format, currAccount.displayName);
+                MessageManager.instance.createMessage(text, STANDARD, currAccount, event.room);
+
+                if (data.isDone()) {
+                    AppEventManager.instance.post(new GroupJoinedEvent(entry.getValue().groupName));
+                    // We're finally done processing this group invitation so remove it from the list
+                    mInvitedGroups.remove(entry.getKey());
+                }
+            }
+        }
+    }
+
+    /** Handle the group profile changes */
+    @Subscribe public void onGroupProfileChange(@NonNull final ProfileGroupChangeEvent event) {
+        // Check for joined groups to see if there are any to report and if the current event
+        // is for an event on our list. If the list is empty, just return.
+        if (!mInvitedGroups.keySet().contains(event.key)) return;
+
+        Account currAccount = AccountManager.instance.getCurrentAccount();
```

Comment:
You probably want to check currAccount for null here.  Android Studio does a great job of generating warnings for these possible NPE conditions, as does the debugger.  I'm pretty sure that currAccount cannot be null in a database handler (due to event bus getting shut down) but better safe than sorry.